### DISCUSSION TITLE
Does cypress now support React 18 by default or is the user still required to opt in via the React 18 adapter?

### DISCUSSION THREAD

**Discussion by longmauriceproj** [cypress-io/cypress#27211] [Questions and Help]
Asking for clarification on [feat: React 18 support #22876](https://github.com/cypress-io/cypress/pull/22876). 

The post mentions that `cypress/react` will support the main version of React in the next major version of Cypress, but I never see it mentioned in the changelogs for subsequent major version releases. As stated in the title, is the user still required to opt in via the React 18 adapter i.e. use `import { mount } from 'cypress/react18'` instead of `import { mount } from 'cypress/react'`?

**Comment by portyu9** [comment]
Hi @longmauriceproj 

There are two different answers depending on Cypress version.

For **Cypress 12.x**, which was current when this discussion was posted, React 18 still used the dedicated adapter:

```js
import { mount } from 'cypress/react18'
```

So your reading of the historical PR was correct: React 18 support existed, but it was still an explicit React 18 adapter at that point.

That has since changed.

With current Cypress, React 18 (and supported newer React versions) uses:

```js
import { mount } from 'cypress/react'
```

The `cypress/react18` harness has been deprecated and is no longer shipped. React 18 support was merged into the normal `cypress/react` test harness.

So for a current project, **do not opt into `cypress/react18`**; use `cypress/react`.

I reviewed the full discussion and the linked React 18 Cypress PR (#22876). There were no screenshot attachments or discussion replies.

Please mark this answer as accepted if it helped, thank you.
